test/internal/cmd/cmd.go

```diff
@@ -0,0 +1,89 @@
+//go:build windows
+
+package cmd
+
+import (
+	"context"
+	"errors"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/opencontainers/runtime-spec/specs-go"
+
+	"github.com/Microsoft/hcsshim/internal/cmd"
+	"github.com/Microsoft/hcsshim/internal/cow"
+	"github.com/Microsoft/hcsshim/internal/log"
+)
+
+const CopyAfterExitTimeout = time.Second
+
+func desc(c *cmd.Cmd) string {
+	desc := "init command"
+	if c.Spec != nil {
+		desc = strings.Join(c.Spec.Args, " ")
+	}
+
+	return desc
+}
+
+func Create(ctx context.Context, _ testing.TB, c cow.ProcessHost, p *specs.Process, io *BufferedIO) *cmd.Cmd {
+	cc := &cmd.Cmd{
+		Host:                 c,
+		Context:              ctx,
+		Spec:                 p,
+		Log:                  log.G(ctx),
+		CopyAfterExitTimeout: CopyAfterExitTimeout,
+		ExitState:            &cmd.ExitState{},
+	}
+	io.AddToCmd(cc)
+
+	return cc
+}
+
+func Start(_ context.Context, t testing.TB, c *cmd.Cmd) {
+	if err := c.Start(); err != nil {
+		t.Helper()
+		t.Fatalf("failed to start %q: %v", desc(c), err)
+	}
+}
+
+func Run(ctx context.Context, t testing.TB, c *cmd.Cmd) int {
+	Start(ctx, t, c)
+	e := Wait(ctx, t, c)
+
+	return e
+}
+
+func Wait(_ context.Context, t testing.TB, c *cmd.Cmd) int {
+	// todo, wait on context.Done
```

Comment:
why did we not do this in this PR?

Reply:
I didnt need it yet, and I didnt want to include a whole new batch of tests to see if canceling contexts work as expected
It was more of a note for when we expand the functional tests